lib/_http_outgoing.js

```diff
@@ -138,15 +138,15 @@ OutgoingMessage.prototype._writeRaw = function(data, encoding, callback) {
       this.connection.writable &&
       !this.connection.destroyed) {
     // There might be pending data in the this.output buffer.
-    while (this.output.length) {
-      if (!this.connection.writable) {
-        this._buffer(data, encoding, callback);
-        return false;
+    var outputLength = this.output.length;
+    if (outputLength) {
+      for (var i = 0; i < outputLength; i++) {
+        var cb = this.outputCallbacks[i];
+        this.connection.write(this.output[i], this.outputEncodings[i], cb);
       }
-      var c = this.output.shift();
-      var e = this.outputEncodings.shift();
-      var cb = this.outputCallbacks.shift();
-      this.connection.write(c, e, cb);
+      this.output = [];
```

Comment:
The previous version didn't create new arrays, instead of `shift()`ed off all the values - basically emptying the array, which for the new version, I simply emptied the array by making a new one.

[JSPerf](http://jsperf.com/length-equal-0-or-new-array/20) says that using `= [];` instead of `.length = 0;` is faster, but if you know differently I'd be open to hearing about it.

Building the arrays in reverse order by using `array.unshift()` would, I think, impact other benchmarks in a bad way to the point of making that optimization a de-optimization though.